src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/TripleDesCngCipher.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using Internal.NativeCrypto;
+
+namespace Internal.Cryptography
+{
+    internal sealed class TripleDesCngCipher : BasicSymmetricCipher
+    {
+        private static readonly SafeAlgorithmHandle s_hAlgCbc = Open3DesAlgorithm(Cng.BCRYPT_CHAIN_MODE_CBC);
+        private static readonly SafeAlgorithmHandle s_hAlgEcb = Open3DesAlgorithm(Cng.BCRYPT_CHAIN_MODE_ECB);
+
+        private readonly bool _encrypting;
+        private SafeKeyHandle _hKey;
+        private byte[] _currentIv;  // CNG mutates this with the updated IV for the next stage on each Encrypt/Decrypt call.
+                                    // The base IV holds a copy of the original IV for Reset(), until it is cleared by Dispose().
+
+        public TripleDesCngCipher(CipherMode cipherMode, int blockSizeInBytes, byte[] key, byte[] iv, bool encrypting)
+            : base(cipherMode.GetCipherIv(iv), blockSizeInBytes)
+        {
+            _encrypting = encrypting;
+
+            if (IV != null)
+            {
+                _currentIv = new byte[IV.Length];
+            }
+
+            SafeAlgorithmHandle hAlg = GetCipherAlgorithm(cipherMode);
+            _hKey = hAlg.BCryptImportKey(key);
+            Reset();
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                SafeKeyHandle hKey = _hKey;
+                _hKey = null;
+                if (hKey != null)
+                {
+                    hKey.Dispose();
+                }
+
+                byte[] currentIv = _currentIv;
+                _currentIv = null;
+                if (currentIv != null)
+                {
+                    Array.Clear(currentIv, 0, currentIv.Length);
+                }
+            }
+
+            base.Dispose(disposing);
```

Comment:
Is any of this shareable with other BasicSymmetricCiphers?  e.g. should there be a helper on the base that provides the cleanup?  Or maybe this is the only derived class at this point and this is a case of "we can refactor later when there's a need" (which would be a fine answer)?